docs/devel/infrastructure-interaction.md

```diff
@@ -0,0 +1,85 @@
+**Problem**
+
+We've got at least 3 cases where kube needs to know about and interact with functionality provided by the underlying infrastructure. The 3 I can think of from the top of my head are:
+
+1. minion discovery
+2. storage (be it a GCE PD, NFS, Ceph, or something else [durable data is a little special])
+3. externally reachable functionality (external IPs, external load balancers, ssl termination points, etc)
+
+We don't seem to have a common pattern how kube and the cloud provider/infrastructure should be working together to expose functionality. I'd like to start a discussion of how I see at least a start of how external functionality and kube can get along.
+
+**The three ways kube and infrastructure interact**
```

Comment:
Hmm. I think I'm using push/pull terminology in the opposite way as this writeup.